### PR TITLE
[0.9.1] Update the documentation on the usage of docker images in the deployments

### PR DESCRIPTION
**Pull request by angrybayblade**
## Proposed changes

Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request.

## Fixes

If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [ ] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have locally run services that could be impacted and they do not present failures derived from my changes

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by Adamantios — COMMENTED** [review]
Looks good, I would suggest that @jmoreira-valory polishes it a bit at some point.

Please fix the tests.

File: docs/advanced_reference/commands/autonomy_build-image.md [hunk 1]
```diff
@@ -2,11 +2,19 @@ Builds the agent image for a service.
 
 The command must be executed within the service directory. That is, a directory containing the service configuration file (`service.yaml`). Labels for images generated using this command will follow the format `valory/oar-<PUBLIC_ID>`.
 
-To build an image for a specific agent without fetching a service, run
+To build an image for a specific agent without fetching a service navigate to the service directory, and run
```
Comment (jmoreira-valory) [inline docs/advanced_reference/commands/autonomy_build-image.md:5]: How can the user navigate to a service repository without fetching the service?

Comment (angrybayblade) [reply docs/advanced_reference/commands/autonomy_build-image.md:5]: My bad =

File: docs/advanced_reference/commands/autonomy_build-image.md [hunk 2]
```diff
@@ -2,11 +2,19 @@ Builds the agent image for a service.
...
+
+```bash
+autonomy build-image
+```
+
+Or if you want to build an image for a specific agent package, run
+
 ```bash
-autonomy build-image PUBLIC_ID_OR_HASH
+autonomy build-image PUBLIC_ID
```
[the remark this answers is not in the document]
Comment (jmoreira-valory) [reply docs/advanced_reference/commands/autonomy_build-image.md:14]: Also, below we indicate PUBLIC_ID_OR_HASH, but now this changed to PUBLIC_ID only... please clarify.

File: docs/advanced_reference/commands/autonomy_build-image.md [hunk 3]
```diff
@@ -43,8 +56,54 @@ autonomy build-image [OPTIONS] [PUBLIC_ID_OR_HASH]
...
+
+```bash
+$ autonomy build-image <author>/<agent_package>:<package_hash> --version <version>
+```
+
+This will tag the image using the `<version>` string rather than the `<package_hash>`, which would be `<author>/oar-<agent_package>:<version>` in this case
+
+### Use custom author name for tagging the image
+
+By default the command will use the author name from the local CLI config stored in `~/.aea` to tag the image. The user can override this using the `--image-author` flag.
```
Comment (jmoreira-valory) [inline docs/advanced_reference/commands/autonomy_build-image.md:92]: Write more clearly where this author comes from... "~/.aea" has not been addressed anywhere. It should refer to the author name specified with autonomy init.

File: docs/advanced_reference/developer_tooling/dev_mode.md [hunk 4]
```diff
@@ -1,6 +1,8 @@
 The {{open_autonomy}} framework comes with *dev mode* tooling to enable faster service developing and debugging. The dev mode supports running agent services with a number of functionalities enabled:
 
-* **Hot reload,** which enables hot code swapping and reflects changes on the agent code as well as on the local `open-aea` repository without rebuilding or restarting the containers manually.
+* **Hot reload** which enables hot code swapping and reflects changes on the agent code as well as on the local `open-aea` repository without rebuilding or restarting the containers manually.
+* **Hardhat Instance** which can be used as a chain to run the service against.
+* **ACN Instance** which can be used to test various agent communication functionalities, for example establishing the tendermint network at the startup.
```
Comment (jmoreira-valory) [inline docs/advanced_reference/developer_tooling/dev_mode.md:5]: Inclusion of a pre-configured **Hardhat Instance** image in the deployment, which can be used as a test bockchain.
Inclusion of a pre-configured **ACN Instance** image, which can be used...

File: docs/advanced_reference/use_custom_images.md [hunk 5]
```diff
@@ -10,14 +10,18 @@ The current deployment setup uses 5 docker images.
...
+    This image contains the deployment setup for a `tendermint` node. When building a deployment, the deployment builder will use `valory/open-autonomy-tendermint` as the name and the current version of the framework as the version tag. If you want to change that you can change the name and the version of the image using `TENDERMINT_IMAGE_NAME` and `TENDERMINT_IMAGE_VERSION` environment variables respectively.
 
-4. `valory/open-autonomy-hardhat` 
+4. `valory/open-autonomy-hardhat`
    
-   This image comes with the Autonolas registry contracts pre deployed. This image should only be used when working with the `dev` mode. When building a deployment, the deployment builder will use `valory/open-autonomy-hardhat` as the name and `latest` as the version tag. If you want to change that you can change the name and the version of the image using `HARDHAT_IMAGE_NAME` and `HARDHAT_IMAGE_VERSION` environment variables respectively.
+   This images contains a pre-configured hardhat node which can be used for testing the services locally and running end to end tests for the agent packages.
```
Comment (angrybayblade) [inline docs/advanced_reference/use_custom_images.md:17]: Formatting needs to be fixed here

File: docs/advanced_reference/use_custom_images.md [hunk 6]
```diff
@@ -1,23 +1,45 @@
...
 
-   If you want to include custom contracts in this image read [here](https://github.com/valory-xyz/autonolas-registries/blob/main/docs/running_with_custom_contracts.md) on how to do it.
 
-5. `valory/open-acn-node` 
-   
-   This image contains the deployment setup for an `ACN` node which can be used for direct agent to agent communication. This image should only be used when working with the `dev` mode. When building a deployment, the deployment builder will use `valory/open-acn-node` as the name and `latest` as the version tag. If you want to change that you can change the name and the version of the image using `ACN_IMAGE_NAME` and `ACN_IMAGE_VERSION` environment variables respectively.
+`valory/oar-<author>/<agent_package>:<package_hash>`
+
+: This image extends the base image `valory/open-autonomy`. It contains the agent package for a service and a deployment environment for the same agent. This image is built through the `autonomy build-image` command.
```
[the remark this answers is not in the document]
Comment (angrybayblade) [reply docs/advanced_reference/use_custom_images.md:12]: This looks like a merge error, I'll fix it

**Comment by Adamantios** [comment]
This PR cannot be for `0.9.1`. #1833 with `[0.10.0]` tag, i.e. breaking changes, has been included here as it is on `main`.

**Review by jmoreira-valory — APPROVED** [review]

File: docs/advanced_reference/developer_tooling/dev_mode.md [hunk 7]
```diff
@@ -40,44 +33,42 @@ Execute the next steps in a separate terminal.
...
+            "private_key": "0x59c6995e998f97a5a0044966f0945389dc9e86dae88c7a8412f4603b6b78690d"
+        },
+        {
+            "address": "0x3C44CdDdB6a900fa2b585dd299e03d12FA4293BC",
+            "private_key": "0x5de4111afa1a4b94908f83103eb1f1706367c2e68ca870fc3fb9a804cdab365a"
+        },
+        {
+            "address": "0x90F79bf6EB2c4f870365E785982E1f101E93b906",
+            "private_key": "0x7c852118294e51e653712a81e05800f419141751be58f605c371e15141b007a6"
+        }
```
Comment (Adamantios) [inline docs/advanced_reference/developer_tooling/dev_mode.md:55]: @angrybayblade did the price oracle run for more than one period with the changes on `docs/oracle-demo`? If yes, let's revert this change.
